### PR TITLE
test: 82% coverage of split expense

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=0PjYGEm)
### <samp>🤖 Generated by Copilot at 2080a4f</samp>

This pull request adds tests and mock data for the split expense page component, which allows users to split an expense by different criteria. It also adds the `CurrencyObj` interface to the models folder, which represents a currency object used in the app.

### <samp>🤖 Generated by Copilot at 2080a4f</samp>

> _We split the expense of doom_
> _With mock data and service stubs_
> _We test the currency object_
> _By percentage, amount and cost_

### <samp>🤖 Generated by Copilot at 2080a4f</samp>

*  Add the CurrencyObj interface to define the currency object model ([link](https://github.com/fylein/fyle-mobile-app/pull/2236/files?diff=unified&w=0#diff-50ed6f7e6b117e54776e5c13b548047851cf227684bc1956d5c2381c9e3fd98cR1-R6))
*  Create a mock data file for the CurrencyObj interface for testing purposes ([link](https://github.com/fylein/fyle-mobile-app/pull/2236/files?diff=unified&w=0#diff-5ea12bdb0fc7d819953dd88adee362a5f47791e5156244f58ba7d3e502e5729aR1-R8))
*  Import the mock data files for custom input, cost centers and currency object to the split expense page spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2236/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R128-R130))
*  Set up the launchDarklyService mock to return false for the getVariation method and the component transaction property to a clone of the txnData mock ([link](https://github.com/fylein/fyle-mobile-app/pull/2236/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R1013-R1014))
  - The page title should be 'Split Expense' ([link](https://github.com/fylein/fyle-mobile-app/pull/2236/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R1074-R1144))

## Clickup
https://app.clickup.com/t/85ztjmt7v

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 1]
```diff
@@ -541,7 +541,6 @@ export class SplitExpensePage implements OnInit {
     );
 
     if (this.splitType === 'cost centers') {
-      const orgSettings$ = this.orgSettingsService.get();
```
Comment (suyashpatil78) [inline src/app/fyle/split-expense/split-expense.page.ts:544]: This call was unnecessary as we are already setting orgSettings$ in `ionViewWillEnter`

**Comment by suyashpatil78** [comment]
@OmkarJ13 @Julias0 @Dimple16 Will fix eslint issues later before merging into master

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 2]
```diff
@@ -1066,6 +1071,77 @@ describe('SplitExpensePage', () => {
...
+        expect(orgSettingsService.get).toHaveBeenCalledTimes(1);
+        expect(orgUserSettingsService.get).toHaveBeenCalledTimes(1);
+        expect(orgUserSettingsService.getAllowedCostCenters).not.toHaveBeenCalled();
+        expect(costCenters).toEqual([]);
+      });
+    });
+
+    it('should set isCorporateCardsEnabled$ correctly and call setValuesForCCC once', () => {
+      activateRouteMock.snapshot.params.currencyObj =
+        '{"amount":23,"currency":"USD","orig_amount":23,"orig_currency":"USD"}';
```
Comment (OmkarJ13) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1134]: should we do put an object and use JSON.stringify here? Would be easier to read and catch errors I think

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 3]
```diff
@@ -1066,6 +1071,77 @@ describe('SplitExpensePage', () => {
...
       });
     });
+
+    it('should set dependentCustomProperties$ correctly if splitType is projects', () => {
+      component.ionViewWillEnter();
+
+      component.dependentCustomProperties$.subscribe((dependentCustomProperties) => {
+        expect(dependentFieldsService.getDependentFieldValuesForBaseField).toHaveBeenCalledOnceWith(
+          txnData.custom_properties,
+          undefined
```
Comment (OmkarJ13) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1081]: Should we also handle the case where the parentFieldId is not `undefined`?

Comment (suyashpatil78) [reply src/app/fyle/split-expense/split-expense.page.spec.ts:1094]: Added tests for that

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 4]
```diff
@@ -1066,6 +1071,89 @@ describe('SplitExpensePage', () => {
...
+      component.dependentCustomProperties$.subscribe((dependentCustomProperties) => {
+        expect(dependentFieldsService.getDependentFieldValuesForBaseField).toHaveBeenCalledOnceWith(
+          txnData.custom_properties,
+          184692
+        );
+        expect(dependentCustomProperties).toEqual(dependentFieldValues);
+      });
+    });
+
+    it('should set dependentCustomProperties$ to NULL if splitType is projects', () => {
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1088]: didn't understand this description, mention the case when it'll set to null explicitly

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 5]
```diff
@@ -1066,6 +1071,89 @@ describe('SplitExpensePage', () => {
...
+    });
+
+    it('should set isCorporateCardsEnabled$ correctly and call setValuesForCCC once', () => {
+      activateRouteMock.snapshot.params.currencyObj = JSON.stringify(currencyObjData1);
+      spyOn(component, 'setValuesForCCC');
+
+      component.ionViewWillEnter();
+
+      component.isCorporateCardsEnabled$.subscribe((isCorporateCardsEnabled) => {
+        expect(isCorporateCardsEnabled).toEqual(true);
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1152]: `toBeTrue();`

**Review by Dimple16 — APPROVED** [review]
minor comments, fix and merge

**Pull request merged**